homeassistant/components/nzbget/__init__.py

```diff
@@ -127,18 +128,50 @@ def __init__(self, hass, api):
         self.status = None
         self.available = True
         self._api = api
+        self.downloads = None
+        self.completed_downloads = []
 
     def update(self):
         """Get the latest data from NZBGet instance."""
 
         try:
             self.status = self._api.status()
+            self.downloads = self._api.history()
+
+            self.check_completed_downloads()
+
             self.available = True
             dispatcher_send(self.hass, DATA_UPDATED)
         except pynzbgetapi.NZBGetAPIException as err:
             self.available = False
             _LOGGER.error("Unable to refresh NZBGet data: %s", err)
 
+    def init_download_list(self):
+        """Initialize download list."""
+        self.downloads = self._api.history()
+        self.completed_downloads = [
+            (x["Name"], x["Category"], x["Status"]) for x in self.downloads
+        ]
+
+    def check_completed_downloads(self):
+        """Check that downloads have completed."""
+
+        actual_completed_downloads = [
+            (x["Name"], x["Category"], x["Status"]) for x in self.downloads
+        ]
+
+        tmp_completed_downloads = list(
+            set(actual_completed_downloads).difference(self.completed_downloads)
+        )
+
+        for download in tmp_completed_downloads:
+            self.hass.bus.fire(
+                "nzbget_downloaded_nzb",
```

Comment:
Maybe a better name?

```suggestion
                "nzbget_download_complete",
```

Reply:
I was matching the event name from the Transmission component (transmission_downloaded_torrent) but agree that this is better :)